lib/rules/media-feature-name-value-whitelist/README.md

```diff
@@ -0,0 +1,65 @@
+# media-feature-name-value-whitelist
+
+Specify a whitelist of allowed media feature and value pairs.
+
+```css
+@media screen and (min-width: 768px) {}
+/**                    ↑        ↑
+ *         These features and these values */
+```
+
+**Caveat:** Media feature names within range or boolean context are ignored.
+
+## Options
+
+```json
+{
+  "max-width": ["Array", "of", "values"],
+  "/regex/": ["/regex/", "non-regex"]
+}
+```
+
+If a media feature is found, only its whitelisted property values are allowed. The rule reports all
```

Comment:
Let's go with the following to be more consistent with `declaration-property-value-whitelist`:

---

## Options

`object`: `{
  "unprefixed-media-feature-name": ["array", "of", "values"]
}`

If a media feature name is found in the object, only its whitelisted values are allowed. If the media feature name is not included in the object, anything goes.

If a name or value is surrounded with `/` (e.g. `"/width$/"`), it is interpreted as a regular expression. For example, `/width$/` will match `max-width` and `min-width`.

---

Reply:
Is this the only rule that uses `object:`? I personally find it atrocious to look at/read in code as well as rendered.  The JSON object is better formatted. 
```json
{
  "max-width": ["Array", "of", "values"],
  "/regex/": ["/regex/", "non-regex"]
}
```
I believe there is another rule that uses this in some form but I'd have to check.

Reply:
> I believe there is another rule that uses this in some form but I'd have to check.

I think it's only the `declaration-property-*-*list` rules that accept objects. I might be wrong, though.

> I personally find it atrocious to look at/read in code as well as rendered.

Yep, I was thinking the same thing. However, if we can't find another example of it done as a JSON object then I suggest we go with the other approach in this PR to keep things consistent.

We should then create a separate issue to address how we can better communicate the options across the board. To be honest, all the option descriptions need a little love and I don't think we ever rationalised the approach we took. It'd be good to look, as part of that issue, into how other tools communicate their options.